shop/src/main/java/ro/msg/learning/shop/controller/ProductServiceController.java

```diff
@@ -0,0 +1,41 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.entity.Product;
+import ro.msg.learning.shop.entity.ProductCategory;
+import ro.msg.learning.shop.mapper.ProductDtoMapper;
+import ro.msg.learning.shop.service.ProductService;
+
+import java.util.UUID;
+
+public class ProductServiceController {
+    @Autowired
+    ProductService productService;
+    @Autowired
+    ProductDtoMapper productDtoMapper;
+
+    @RequestMapping(value="/products")
+    public ResponseEntity<Object> getProduct(){
+        return new ResponseEntity<>(productService.getProducts(), HttpStatus.OK);
+    }
+    @PutMapping(value = "/products/{id}")
+    public ResponseEntity<Object>
+    updateProduct(@PathVariable("id") UUID id, @RequestBody Product product) {
+
+        productService.updateProduct(id, product);
+        return new ResponseEntity<>("Product is updated successfully", HttpStatus.OK);
+    }
+    @DeleteMapping(value = "/products/{id}")
+    public ResponseEntity<Object> delete(@PathVariable("id") UUID id) {
+        productService.deleteProduct(id);
+        return new ResponseEntity<>("Product is deleted successfully", HttpStatus.OK);
+    }
+    @PostMapping(value = "/products")
+    public ResponseEntity<Object> createProduct(@RequestBody Product product, ProductCategory productCategory) {
```

Comment:
Why use the generic Object when you can return ResponseEntity<ProductDTO>? Same thing for every method in Controller, so at Controller level we deal with DTOs and at Service level we deal with Entities as we discussed. Mappings will also be at Controller level.